File: src/Entity/Poll.php
<?php

namespace App\Entity;

use Doctrine\ORM\Mapping as ORM;

use App\Entity\Chat;
use Doctrine\Common\Collections\Collection;
use Doctrine\Common\Collections\ArrayCollection;
use App\Entity\PollOption;
use App\Entity\PollVote;

#[ORM\Entity]
class Poll
{

    #[ORM\Id]
    #[ORM\Column(type: "integer")]
    private int $id;

    #[ORM\ManyToOne(targetEntity: Chat::class, inversedBy: "polls")]
    #[ORM\JoinColumn(name: 'chat_id', referencedColumnName: 'id', onDelete: 'CASCADE')]
    private Chat $chat_id;

    #[ORM\Column(type: "string", length: 255)]
    private string $question;

    #[ORM\Column(type: "boolean")]
    private bool $is_closed;

    #[ORM\Column(type: "datetime")]
    private \DateTimeInterface $created_at;

    #[ORM\OneToMany(mappedBy: "poll_id", targetEntity: PollOption::class)]
    private Collection $pollOptions;

    #[ORM\OneToMany(mappedBy: "poll_id", targetEntity: PollVote::class)]
    private Collection $poll_votes;

    public function __construct()
    {
        $this->pollOptions = new ArrayCollection();
        $this->poll_votes = new ArrayCollection();
    }

    public function getId()
    {
        return $this->id;
    }

    public function setId($value)
    {
        $this->id = $value;
    }

    public function getChat_id()
    {
        return $this->chat_id;
    }

    public function setChat_id($value)
    {
        $this->chat_id = $value;
    }

    public function getQuestion()
    {
        return $this->question;
    }

    public function setQuestion($value)
    {
        $this->question = $value;
    }

    public function getIs_closed()
    {
        return $this->is_closed;
    }

    public function setIs_closed($value)
    {
        $this->is_closed = $value;
    }

    public function getCreated_at()
    {
        return $this->created_at;
    }

    public function setCreated_at($value)
    {
        $this->created_at = $value;
    }

    public function getPollOptions(): Collection
    {
        return $this->pollOptions;
    }

    public function addPollOption(PollOption $pollOption): self
    {
        if (!$this->pollOptions->contains($pollOption)) {
            $this->pollOptions[] = $pollOption;
            if ($pollOption->getPoll_id() !== $this) {
                $pollOption->setPoll_id($this);
            }
        }

        return $this;
    }

    public function removePollOption(PollOption $pollOption): self
    {
        if ($this->pollOptions->removeElement($pollOption)) {
            if ($pollOption->getPoll_id() === $this) {
                $pollOption->setPoll_id(null);
            }
        }

        return $this;
    }

    public function getPoll_votes(): Collection
    {
        return $this->poll_votes;
    }

    public function addPollVote(PollVote $pollVote): self
    {
        if (!$this->poll_votes->contains($pollVote)) {
            $this->poll_votes[] = $pollVote;
            if ($pollVote->getPoll_id() !== $this) {
                $pollVote->setPoll_id($this);
            }
        }

        return $this;
    }

    public function removePollVote(PollVote $pollVote): self
    {
        if ($this->poll_votes->removeElement($pollVote)) {
            if ($pollVote->getPoll_id() === $this) {
                $pollVote->setPoll_id(null);
            }
        }

        return $this;
    }
}

File: OneDrive/Bureau/Hackify-integrationn/src/main/java/models/Poll.java
package models;

import java.sql.Timestamp;
import java.util.List;

public class Poll {
    private int id;
    private int chatId; // Foreign key to chat
    private String question; // Poll question
    private List<PollOption> options; // List of poll options
    private boolean isClosed; // Whether the poll is closed
    private Timestamp createdAt; // Creation timestamp

    // Default constructor
    public Poll() {
    }

    // Constructor for creating a new poll
    public Poll(int chatId, String question, List<PollOption> options) {
        this.chatId = chatId;
        this.question = question;
        this.options = options;
        this.isClosed = false;
        this.createdAt = new Timestamp(System.currentTimeMillis());
    }

    // Constructor for retrieving from database
    public Poll(int id, int chatId, String question, boolean isClosed, Timestamp createdAt) {
        this.id = id;
        this.chatId = chatId;
        this.question = question;
        this.isClosed = isClosed;
        this.createdAt = createdAt;
    }

    // Full constructor including options
    public Poll(int id, int chatId, String question, List<PollOption> options, boolean isClosed, Timestamp createdAt) {
        this.id = id;
        this.chatId = chatId;
        this.question = question;
        this.options = options;
        this.isClosed = isClosed;
        this.createdAt = createdAt;
    }

    // Getters and setters
    public int getId() {
        return id;
    }

    public void setId(int id) {
        this.id = id;
    }

    public int getChatId() {
        return chatId;
    }

    public void setChatId(int chatId) {
        this.chatId = chatId;
    }

    public String getQuestion() {
        return question;
    }

    public void setQuestion(String question) {
        this.question = question;
    }

    public List<PollOption> getOptions() {
        return options;
    }

    public void setOptions(List<PollOption> options) {
        this.options = options;
    }

    public boolean isClosed() {
        return isClosed;
    }

    public void setClosed(boolean isClosed) {
        this.isClosed = isClosed;
    }

    public Timestamp getCreatedAt() {
        return createdAt;
    }

    public void setCreatedAt(Timestamp createdAt) {
        this.createdAt = createdAt;
    }
}